src/abstractdataframe/selection.jl

```diff
@@ -514,22 +532,71 @@ julia> select(df, AsTable(:) => ByRow(mean))
 ```
 
 """
-select(df::DataFrame, args::AbstractVector{Int}; copycols::Bool=true) =
+select(df::AbstractDataFrame, args...; copycols::Bool=true) =
+    _manipulate(df, args..., copycols=copycols, keeprows=true)
+
+"""
+    transform(df::AbstractDataFrame, args...; copycols::Bool=true)
+
+Create a new data frame that contains columns from `df` and adds columns
+specified by `args` and return it.
+The result is guaranteed to have the same number of rows as `df`.
+Equivalent to `select(df, :, args..., copycols=copycols)`.
```

Comment:
I would remove it actually — I don’t think this is particularly helpful.

Reply:
But what do you propose to add instead? We should give some precise specification how the function works.